src/libraries/illink-sharedframework.targets

```diff
@@ -13,26 +13,16 @@
       <ILLinkArgs>$(ILLinkArgs) -c link</ILLinkArgs>
       <!-- update debug symbols -->
       <ILLinkArgs>$(ILLinkArgs) -b true</ILLinkArgs>
-      <!-- suppress warnings with the following codes:
-           IL2006: The generic parameter 'T' from A with dynamically accessed member kinds B is passed into the generic parameter
-                   'T' from 'System.Lazy<T>' which requires dynamically accessed member kinds 'PublicParameterlessConstructor'
-           IL2009: Could not find method A in type B specified in resource C
-           IL2025: Duplicate preserve of A in B
-           IL2026: Calling A which has B can break functionality when trimming application code. The target method might be removed.
-           IL2035: Unresolved assembly A in DynamicDependencyAttribute on B
-           IL2050: P/invoke method A declares a parameter with COM marshalling. Correctness of COM interop
-                   cannot be guaranteed after trimming. Interfaces and interface members might be removed.
-      -->
-      <LinkerNoWarn>IL2006;IL2009;IL2025;IL2026;IL2035;IL2050</LinkerNoWarn>
+      <!-- suppress IL2025: Duplicate preserve of A in B -->
+      <LinkerNoWarn>IL2025</LinkerNoWarn>
```

Comment:
Do you happen to know where `IL2025` is coming from? I would hope we could have no global suppressions by default.

Reply:
They are all coming from analysis on the embedded `System.Private.CoreLib.xml` file in the input `System.Private.CoreLib` assembly - https://gist.github.com/layomia/2f2d007b9bac71c32fa0aff0e5832ec8. I'm not entirely sure - does this mean that we can remove some of the entries in the CoreLib descriptor XML file due to linker annotations present in the various libraries? Will reach out offline to discuss.

Reply:
https://github.com/dotnet/runtime/pull/40691#discussion_r497097315